app/settlement/update-settlement-status.js

```diff
@@ -1,12 +1,22 @@
 const db = require('../data')
 
-const updateSettlementStatus = async (settlement) => {
-  const updated = await db.completedPaymentRequest.update({
+const updateSettlementStatus = async (settlement, filter) => {
+  const completedPaymentRequest = await db.completedPaymentRequest.findOne({
+    where: {
+      ...filter
+    }
+  })
+
+  if (!completedPaymentRequest) {
+    return undefined
+  }
```

Comment:
Breaking this up into two stages to resolve the issue where we get notified of an unmatched settlement when the settlement has been matched, it just doesn't need updating.

1. is there a matching payment request, if no then we need an alert to investigate
2. if there is update the settlement status IF it needs updating

If 2, then return the invoice number and FRN that can be passed to the events.  Can no longer rely on this from the settlement file as they're not included for these schemes.

Reply:
Yes it was.  Because we were handling "does it exist" and "does it need updating" in one query.  If either of those were false, then we'd get error saying it didn't exist when that only applies to the first check.

For the second part I meant that all existing settlements up to now include an FRN and an Invoice Number.  The event we send for successful update just pulls those values out of the settlement.

However, with these additional schemes they don't necessarily have those properties that the events need.  So if the request is matched, instead of just returning true, we can return the FRN and Invoice Number from the payment request it was matched to and pass that to the events functions.

Reply:
Oh okay - yep